### PR TITLE
Move CombineErrors to consumererror package

### PR DESCRIPTION
**Pull request by bogdandrutu**
Fixes https://github.com/open-telemetry/opentelemetry-collector/issues/2472

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: receiver/jaegerreceiver/trace_receiver.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ import (
 	"go.opentelemetry.io/collector/component/componenterror"
```
Comment (jpkrohling) [inline receiver/jaegerreceiver/trace_receiver.go:49]: Is this still being used somewhere?

**Review by jpkrohling — COMMENTED** [review]
Is there an issue tracking this change? What's the idea behind it? I particularly find `componenterror` more intuitive than `consumererror`, but I might be missing some context here.

**Review by carlosalberto — APPROVED** [review]

**Comment by bogdandrutu** [comment]
A Component is a Consumer + few extra functions. The part of the component that uses this it the Consumer, and there are other pure Consumers which use this.

**Comment by bogdandrutu** [comment]
@jpkrohling good point on the issue. I will open one explaining why some errors should be consumer some should be component. We don't have a clear logic

**Comment by bogdandrutu** [comment]
@jpkrohling good point on the issue. I will open one explaining why some errors should be consumer some should be component. We don't have a clear logic

File: receiver/jaegerreceiver/trace_receiver.go (hunk 1 above)
Comment (bogdandrutu) [reply receiver/jaegerreceiver/trace_receiver.go:49]: Yes for the constant errors `ErrAlreadyStarted`, etc. defined in that package

**Comment by bogdandrutu** [comment]
Issue created.

**Review by tigrannajaryan — COMMENTED** [review]
It would help to prepare in advance a diagram of dependencies that we want to end up with to make reviewing PRs like this easier. I have a hard time keeping the desired picture in my mind.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**